src/index.js

```diff
@@ -1,5 +1,19 @@
+import { AppContainer } from 'react-hot-loader';
 import React from 'react';
 import ReactDOM from 'react-dom';
 import App from './App';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+const rootEl = document.getElementById('root');
+ReactDOM.render(
+  <AppContainer component={App} />,
+  rootEl
+);
+
+if (module.hot) {
+  module.hot.accept('./App', () => {
+    ReactDOM.render(
+      <AppContainer component={require('./App').default} />,
```

Comment:
For educational purposes, does webpack2 automagically update the reference stored in a variable via import after a hot reload?